cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -186,116 +194,108 @@ func run(ctx context.Context, target, assetType, optJSON string, state checkstat
-	// name and version.
-	sort.Slice(opvArr, func(i, j int) bool {
-		si := getScore(opvArr[i].severity)
-		sj := getScore(opvArr[j].severity)
-		switch {
-		case si != sj:
-			return si > sj
-		case opvArr[i].name != opvArr[j].name:
-			return opvArr[i].name < opvArr[j].name
-		default:
-			return opvArr[i].version < opvArr[j].version
+		for key, l := range outdatedPackageVulns {
+			// Sort outdated packages by severity
+			sort.Slice(l, func(i, j int) bool {
+				return getScore(l[i].severity) > getScore(l[j].severity)
+			})
+			// Keep only top vulnTruncateLimit
+			if len(l) > vulnTruncateLimit {
+				logger.Warnf("truncate to top %d vulnerabilities\n", vulnTruncateLimit)
+				outdatedPackageVulns[key] = l[0:vulnTruncateLimit]
+			}
 		}
-	})
-
-	// To avoid report size overflow only top 30 most vulnerable packages
-	// are reported.
-	totalVulnerablePackages := len(opvArr)
-	if totalVulnerablePackages > vulnTruncateLimit {
-		logger.Warnf("truncate to top %d vulnerabilities\n", vulnTruncateLimit)
-		opvArr = opvArr[0:vulnTruncateLimit]
-	}
 
-	vp := report.ResourcesGroup{
-		Name: "Package Vulnerabilities",
-		Header: []string{
-			"Name",
-			"Version",
-			"FixedBy",
-			"Vulnerabilities",
-		},
-	}
-	for _, op := range opvArr {
-		affectedResource := fmt.Sprintf("%s-%s", op.name, op.version)
-		fingerprint := helpers.ComputeFingerprint(op.severity, op.cves)
-
-		// Build vulnerabilities Rsources table.
-		vResourcesTable := make(map[string]string)
-		vResourcesTable["Name"] = op.name
-		vResourcesTable["Version"] = op.version
-		vResourcesTable["FixedBy"] = op.fixedBy
-
-		for i := 0; i < len(op.cves) && i < vulnCVETrucateLimit; i++ {
-			vResourcesTable["Vulnerabilities"] = fmt.Sprintf("%s | [%s](https://nvd.nist.gov/vuln/detail/%s)", vResourcesTable["Vulnerabilities"], op.cves[i], op.cves[i])
-		}
-		if len(op.cves) > vulnCVETrucateLimit {
-			logger.Warnf("truncate affected package [%s] CVE list to [%d]\n", op.name, vulnCVETrucateLimit)
-			vResourcesTable["Vulnerabilities"] = fmt.Sprintf("%s | and some others ...)", vResourcesTable["Vulnerabilities"])
+		vp := report.ResourcesGroup{
+			Name: "Package Vulnerabilities",
+			Header: []string{
+				"FixedBy",
+				"CVE",
+				"Severity",
+				"CWEs",
+				"Link",
+			},
 		}
+		for key, l := range outdatedPackageVulns {
+			vp.Rows = []map[string]string{}
+			maxScore := getScore("NONE")
+			for i, l := range l {
+				if i > vulnCVETrucateLimit {
+					break
+				}
+				newScore := getScore(l.severity)
+				if newScore > maxScore {
+					maxScore = newScore
+				}
+				row := make(map[string]string, len(vp.Header))
+				row["FixedBy"] = l.fixedBy
+				if l.cves != nil {
+					row["CWEs"] = strings.Join(l.cves, ",")
+				}
+				row["CVE"] = l.cve
+				if l.link == "" {
+					l.link = fmt.Sprintf("https://nvd.nist.gov/vuln/detail/%s", l.cve)
+				}
+				row["Link"] = l.link
+				row["Severity"] = l.severity
+				vp.Rows = append(vp.Rows, row)
+			}
 
```

Comment:
Should we keep this or compute from the from CWES? What if more than one?

Reply:
Sure, below we are hardcoding CWEID:937, but trivy provides CweIDs as an array.
I m wondering if we should compute the final CWEID from the ones provided by Trivy or keep the 937